packages/core/lib/cliHelp.js

```diff
@@ -0,0 +1,33 @@
+const handleHelpInput = function ({ inputStrings }) {
+  let displayHelp = false;
+  //User only enter truffle with no commands, let's show them what's available.
+  if (inputStrings.length === 0) {
+    displayHelp = true;
+  }
+  // handle 'truffle help' and 'truffle --help'
+  if (
+    inputStrings.length === 1 &&
+    (inputStrings[0] === "help" || inputStrings[0] === "--help")
+  ) {
+    displayHelp = true;
+  }
+
+  //if `--help` is in the input, validate and transform the input
+  //in order to give the user help
+  if (inputStrings.some(inputString => ["--help"].includes(inputString))) {
+    //check where --help is used, mutate argument into something the rest
+    //of Truffle can deal with
+    const helpIndex = inputStrings.indexOf("--help");
+
+    if (helpIndex !== -1) {
```

Comment:
This logic can be simplified if you rely on `string.indexOf` to determine if `--help` exists as well as its index
```
  const helpIndex = inputStrings.indexOf("--help");
  if (helpIndex !== -1) {
    // ...
  }
```

Reply:
I'm not sure I understand what you want to change here. You wrote the exact same thing as the code itself.

Reply:
Removing the condition: `if (inputStrings.some(inputString => ["--help"].includes(inputString)))` simplifies the logic. This works because `Array.some()` and `Array.indexOf()` can both test if an item exists in the collection. 

```javascript
    const helpIndex = inputStrings.indexOf("--help");
    if (helpIndex !== -1) {
      inputStrings.splice(helpIndex, 1);
      inputStrings.unshift("help");
    }
    return { displayHelp };
```

Reply:
Ah yes, that makes sense. We should only check once. Gotcha!